CODEOWNERS: cleanup and remove duplicates

nashif opened the pull request:
Remove anything that would be covered by MAINTAINERS.yml leaving only
mostly instances that are not covered.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
CODEOWNERS: cleanup and remove duplicates

Remove anything that would be covered by MAINTAINERS.yml leaving only
mostly instances that are not covered.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

aescolar commented on CODEOWNERS:
```diff
@@ -168,32 +115,20 @@
-/boards/nios2/altera_max10/               @nashif
 /boards/arm/stm32_min_dev/                @sidcha
-/boards/posix/                            @aescolar @daor-oti
 /boards/posix/nrf52_bsim/                 @aescolar @wopu-ot
```
also covered by maintainers

aescolar reviewed (dismissed):
All lines that relate to my areas are good

henrikbrixandersen reviewed (dismissed):
Looks good. Perhaps we should add a note to the top of this file pointing people towards the maintainers file instead?

nashif committed:
doc: clarify role of CODEOWNERS and MAINTAINERS files

Change docs to use MAINTAINERS file as the main file for managing code
areas and 'ownership'.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

fabiobaltieri reviewed (dismissed):
I think that few non-maintainer contributors use the file to get cc'd on changes in specific drivers they care about. Should we support that use case or clarify that it's not really supported and we should only use the MAINTAINERS file?

fabiobaltieri commented on doc/contribute/guidelines.rst:
```diff
@@ -651,8 +651,8 @@ workflow here:
-#. GitHub will assign one or more suggested reviewers (based on the
-   CODEOWNERS file in the repo). If you are a project member, you can
+#. A bot will assign one or more suggested reviewers (based on the
+   MAINTAINETRS file in the repo). If you are a project member, you can
```
typo (MAINTAINERS)

aescolar commented:
> I think that few non-maintainer contributors use the file to get cc'd on changes in specific drivers they care about.

In the original MAINTAINERS file proposal ( https://github.com/zephyrproject-rtos/zephyr/pull/20591 ) we had an "inform" entry to cover this use case. I'd say we should just bring it back.

fabiobaltieri commented:
> In the original MAINTAINERS file proposal ( #20591 ) we had an "inform" entry to cover this use case. I'd say we should just bring it back.

I don't think it would cover that specific use case, my impression is that those users would want to get looped in for some specific drivers they care about, but not an entire area of subsystem. Honestly I can't really think of a clean way of doing this, having such use case covered by changing a file in the repository is going to be a big mess (as the codeowner file is right now), we may as well not support that use case. Gerrit had the concept of "Watched Projects" that I think covers this use case without messing with the repository itself but I doubt GitHub supports such a thing.

nashif commented:
> I suggest sending a mail to zephyr-devel about the change in CODEOWNERS.

sure.
